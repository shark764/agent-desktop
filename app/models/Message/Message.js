/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import Immutable from 'immutable';

export default class Message extends Immutable.Record({
  type: '',
  from: '',
  text: '',
  timestamp: new Date().toISOString(),
  unread: false,
}) {
  constructor({ type, from, text, timestamp, unread }) {
    let messageFrom;
    let messageText;
    if (type == null) {
      throw new Error('type is required');
    }
    if (from == null) {
      messageFrom = '';
      console.warn('from was not found', {
        type,
        from,
        text,
        timestamp,
        unread,
      });
    } else {
      messageFrom = from;
    }
    if (text == null) {
      messageText = '';
      console.warn('text was not found', {
        type,
        from,
        text,
        timestamp,
        unread,
      });
    } else {
      messageText = text;
    }
    if (timestamp == null) {
      throw new Error('timestamp is required');
    }
    if (unread == null) {
      throw new Error('unread is required');
    }
    super({ type, from: messageFrom, text: messageText, timestamp, unread });
  }
}
