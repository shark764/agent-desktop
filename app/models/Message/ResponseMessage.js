/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import Message from './Message';

export default class ResponseMessage extends Message {
  constructor({ id, metadata, from, type, body, timestamp }) {
    const text = body ? body.text : undefined;
    let messageType;
    let messageFrom;

    if (metadata && metadata.type === 'agent') {
      messageType = 'agent';
      messageFrom = from;
    } else if (metadata && metadata.type == null) {
      /* CXV1-20378 We are getting messages coming from system (send SMS action) that are missing values in metadata
      with this solution we avoid throwing error because of null/undefined values */

      messageType = 'system';
      messageFrom = 'system';
    } else {
      messageType = metadata != null ? metadata.type : type;
      messageFrom =
        metadata != null && metadata.name != null ? metadata.name : from;
    }
    super({ id, type: messageType, from: messageFrom, text, timestamp });
  }
}
