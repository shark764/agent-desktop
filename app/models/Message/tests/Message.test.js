/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import Message from '../Message';

describe('Message', () => {
  describe('constructor', () => {
    beforeAll(() => {
      global.console = { warn: jest.fn() };
    });
    it('should construct properly', () => {
      const message = new Message({
        type: 'type',
        from: 'from',
        text: 'text',
        timestamp: 'timestamp',
        unread: false,
      });
      expect(message).toMatchSnapshot();
    });
    it('should error when "type" is not passed in', () => {
      let error;
      try {
        // eslint-disable-next-line no-new
        new Message({
          // type: 'type',
          from: 'from',
          text: 'text',
          timestamp: 'timestamp',
          unread: false,
        });
      } catch (e) {
        error = e;
      }
      expect(error.message).toEqual('type is required');
    });
    it('should set "from" to empty string when "from" is not passed in', () => {
      const message = new Message({
        // eslint-disable-line no-new
        type: 'type',
        // from: 'from',
        text: 'text',
        timestamp: 'timestamp',
        unread: false,
      });
      expect(message).toMatchSnapshot();
    });
    it('should set "text" to empty when "text" is not passed in', () => {
      // eslint-disable-next-line no-new
      const message = new Message({
        type: 'type',
        from: 'from',
        // text: 'text',
        timestamp: 'timestamp',
        unread: false,
      });
      expect(message).toMatchSnapshot();
    });
    it('should error when "timestamp" is not passed in', () => {
      let error;
      try {
        // eslint-disable-next-line no-new
        new Message({
          type: 'type',
          from: 'from',
          text: 'text',
          // timestamp: 'timestamp',
          unread: false,
        });
      } catch (e) {
        error = e;
      }
      expect(error.message).toEqual('timestamp is required');
    });
    it('should error when "unread" is not passed in', () => {
      let error;
      try {
        // eslint-disable-next-line no-new
        new Message({
          type: 'type',
          from: 'from',
          text: 'text',
          timestamp: 'timestamp',
          // unread: false,
        });
      } catch (e) {
        error = e;
      }
      expect(error.message).toEqual('unread is required');
    });
  });
});
