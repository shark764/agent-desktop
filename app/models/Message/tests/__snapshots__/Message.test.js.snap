// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message constructor should construct properly 1`] = `
Object {
  "from": "from",
  "text": "text",
  "timestamp": "timestamp",
  "type": "type",
  "unread": false,
}
`;

exports[`Message constructor should set "from" to empty string when "from" is not passed in 1`] = `
Object {
  "from": "",
  "text": "text",
  "timestamp": "timestamp",
  "type": "type",
  "unread": false,
}
`;

exports[`Message constructor should set "text" to empty when "text" is not passed in 1`] = `
Object {
  "from": "from",
  "text": "",
  "timestamp": "timestamp",
  "type": "type",
  "unread": false,
}
`;
